openmrs-client/src/main/java/org/openmrs/mobile/activities/settings/SettingsFragment.kt

```diff
@@ -0,0 +1,191 @@
+            val typeface = Typeface.createFromAsset(activity!!.assets, "fonts/Roboto/Roboto-Medium.ttf")
+            dismissButton.typeface = typeface
+            dismissButton.setOnClickListener { snackbar.dismiss() }
+            snackBarLayout.addView(customSnackBarView, 0)
+            snackbar.show()
+        } else {
+            binding.conceptsDownloadButton.isEnabled = false
+        }
+        binding.conceptsCountTextView.text = text
+    }
+
+    override fun addLogsInfo(logSize: Long, logFilename: String?) {
+        binding.logsDesc1TextView.text = logFilename
+        binding.logsDesc2TextView.text = context!!.getString(R.string.settings_frag_size) + logSize + "kB"
+        binding.logsLayout.setOnClickListener { view: View ->
+            val i = Intent(view.context, LogsActivity::class.java)
+            startActivity(i)
+        }
+    }
+
+    override fun setUpConceptsView() {
+        with(binding) {
+            languageApplyButton.setOnClickListener { activity!!.recreate() }
+            conceptsDownloadButton.setOnClickListener {
+                conceptsDownloadButton.isEnabled = false
+                val startIntent = Intent(activity, ConceptDownloadService::class.java)
+                startIntent.action = ApplicationConstants.ServiceActions.START_CONCEPT_DOWNLOAD_ACTION
+                Objects.requireNonNull(activity)!!.startService(startIntent)
+            }
+        }
+    }
+
+    override fun addBuildVersionInfo() {
+        var versionName = ""
+        var buildVersion = 0
+        val packageManager = this.activity!!.packageManager
+        val packageName = this.activity!!.packageName
+        try {
+            versionName = packageManager.getPackageInfo(packageName, 0).versionName
+            val ai = packageManager.getApplicationInfo(packageName, PackageManager.GET_META_DATA)
+            buildVersion = ai.metaData.getInt("buildVersion")
+        } catch (e: PackageManager.NameNotFoundException) {
+            mPresenter?.logException("Failed to load meta-data, NameNotFound: " + e.message)
+        } catch (e: NullPointerException) {
+            mPresenter?.logException("Failed to load meta-data, NullPointer: " + e.message)
+        }
+        with(binding) {
+            appNameTextView.text = resources.getString(R.string.app_name)
+            versionTextView.text = versionName + context?.getString(R.string.frag_settings_build) + buildVersion
+        }
+    }
+
+    override fun addPrivacyPolicyInfo() {
+        binding.privacyPolicyLayout.setOnClickListener { view: View ->
+            val i = Intent(Intent.ACTION_VIEW)
+            i.data = Uri.parse(view.context.getString(R.string.url_privacy_policy))
+            startActivity(i)
+        }
+    }
+
+    override fun rateUs() {
+        binding.rateUsLayout.setOnClickListener {
+            val uri = Uri.parse("market://details?id=" + ApplicationConstants.PACKAGE_NAME)
```

Comment:
Use string template literals.
```suggestion
            val uri = Uri.parse("market://details?id=${ApplicationConstants.PACKAGE_NAME}")
```
(not tested code)